src/authentication/register.js

```diff
@@ -0,0 +1,50 @@
+const jwt = require( 'jsonwebtoken' )
+const crypto = require( 'crypto' )
+const User = require( '../../models/users' )
+
+const generateToken = user => {
```

Comment:
Refactor all of the code duplicated between this file and `login.js` into its own module, so that we are not duplicating functionality.